active_learning_test/rb_streams.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import time
+from collections import defaultdict
+from enum import Enum
+from itertools import chain, islice
+from typing import Iterable, Optional, Union
+
+import rubrix as rb
+from rubrix.client.models import Record
+from rubrix.client.sdk.commons.errors import NotFoundApiError
+
+
+class Priority(Enum):
+
+    Critical = 1
+    High = 5
+    Medium = 10
+    Low = 30
+
+
+def __batch_iterable__(iterable, size):
+    sourceiter = iter(iterable)
+    while True:
+        batchiter = islice(sourceiter, size)
+        yield chain([next(batchiter)], batchiter)
+
+
+class DatasetQueryStream:
```

Comment:
I really like this solution. Much cleaner than before. Just for my understanding: We are waiting for the upcoming callbacks in rubrix, otherwise we could dropt the `time.sleep()` here as well, right?